app/routes.py

```diff
@@ -1 +1,298 @@
+               "token": token
+               }
+        response = requests.post(url, data = data)
+
+
+        # this is the part with slack module:
+        # message = f"Someone just completed a task {task.title}"
+        # client = WebClient(token=os.environ.get("SLACKBOT_TOKEN"))
+        # response = client.chat_postMessage(channel="task-notifications", text=message)
+
+         
+    return jsonify({"task":task.to_dict()}), 200
+
+
+# mark_imcomplete
+@task_bp.route("/<task_id>/mark_incomplete", methods = ["PATCH"])
+def mark_imcomplete(task_id):
+    task = Task.query.get(task_id)
+    if not task:
+        return jsonify({"msg": "task not found"}), 404
+    
+    task.completed_at = None
+    task.is_complete = False
+
+    db.session.commit()
+
+    return jsonify({"task":task.to_dict()}), 200
+
+
+
+
+# GOAL
+goal_bp = Blueprint("goal", __name__,url_prefix = "/goals")
+
+# wave 5
+# create a goal
+@goal_bp.route("", methods = ["POST"])
+def create_goal():
+    
+    response_body = request.get_json()
+    if "title" not in response_body:
+        return jsonify({"details": "Invalid data"}), 400
+    new_goal = Goal.from_goal_dict(response_body)
+    
+    db.session.add(new_goal)
+    db.session.commit()
+  
+    return jsonify({
+                    "goal":{
+                    "id":new_goal.goal_id,
+                    "title":new_goal.title
+    }}), 201
+
+
+# helper function
+def validate_goal(model,goal_id):
+    try:
+        goal_id = int(goal_id)
+    except ValueError:
+        return abort(make_response({"msg": "id is invalide input"}, 400))
+    
+    # return Task.query.get_or_404(task_id_num)
+ 
+    goal = model.query.get(goal_id)
+    if goal is None:
+        abort(make_response({"msg": "Goal not found"}, 404))
+    return goal
+
+
+# get goals
+@goal_bp.route("", methods = ["GET"])
+def get_all_goals():
+    response = []
+    goals = Goal.query.all()
+    for goal in goals:
+        response.append(goal.goal_to_dict())
+    return jsonify(response), 200
+
+
+# get one goal
+@goal_bp.route("/<goal_id>", methods = ["GET"])
+def get_one_goal(goal_id):
+    goals = validate_goal(Goal,goal_id)
+
+    return jsonify({"goal":{
+            "id": goals.goal_id,
+            "title": goals.title,
+            }}), 200
+
+
+# update goal
+@goal_bp.route("/<goal_id>", methods = ["PUT"])
+def update_goal(goal_id):
+    goal = validate_goal(Goal,goal_id)
+    request_data = request.get_json()
+    
+    goal.title = request_data["title"]
+
+    db.session.commit()
+
+    return jsonify({"goal":{
+            "id": goal.goal_id,
+            "title": goal.title
+            }
+            }), 200
+
+
+# delete goal
+@goal_bp.route("/<goal_id>", methods = ["DELETE"])
+def delete_goal(goal_id):
+    goal = validate_goal(Goal,goal_id)
+
+    db.session.delete(goal)
+    db.session.commit()
+
+    return {"details": f'Goal 1 "Build a habit of going outside daily" successfully deleted'}
```

Comment:
This should not be hard-coded and should instead print the details about the specific goal that was deleted.